sonic-xcvrd/xcvrd/sff_mgr.py

```diff
@@ -0,0 +1,393 @@
+                # in previous port_dict sanpshot
+                if lport not in self.port_dict_prev or 'type' not in self.port_dict_prev[lport]:
+                    xcvr_inserted = True
+                # Check if there's an diff between current and previous host_tx_ready
+                if (lport not in self.port_dict_prev or
+                        'host_tx_ready' not in self.port_dict_prev[lport] or
+                        self.port_dict_prev[lport]['host_tx_ready'] != data['host_tx_ready']):
+                    host_tx_ready_changed = True
+                # Only proceed on xcvr insertion case or the case of
+                # host_tx_ready getting changed.
+                # Even though handle_port_update_event() filters out duplicate
+                # events, we still need to have this below extra check to ignore
+                # the event that is not related to insertrion or host_tx_ready
+                # change, such as CONFIG_DB related change.
+                if (not xcvr_inserted) and (not host_tx_ready_changed):
+                    continue
+                self.log_notice("{}: xcvr_inserted={}, host_tx_ready_changed={}".format(
+                    lport, xcvr_inserted, host_tx_ready_changed))
+
+                # double-check the HW presence before moving forward
+                sfp = self.platform_chassis.get_sfp(pport)
+                if not sfp.get_presence():
+                    self.log_error("{}: module not present!".format(lport))
+                    del self.port_dict[lport]
+                    continue
+                try:
+                    # Skip if XcvrApi is not supported
+                    api = sfp.get_xcvr_api()
+                    if api is None:
+                        self.log_error(
+                            "{}: skipping sff_mgr since no xcvr api!".format(lport))
+                        continue
+
+                    # Skip if it's not a paged memory device
+                    if api.is_flat_memory():
+                        self.log_notice(
+                            "{}: skipping sff_mgr for flat memory xcvr".format(lport))
+                        continue
+
+                    # Skip if it's a copper cable
+                    if api.is_copper():
+                        self.log_notice(
+                            "{}: skipping sff_mgr for copper cable".format(lport))
+                        continue
+
+                    # Skip if tx_disable action is not supported for this xcvr
+                    if not api.get_tx_disable_support():
+                        self.log_notice(
+                            "{}: skipping sff_mgr due to tx_disable not supported".format(
+                                lport))
+                        continue
+                except AttributeError:
+                    # Skip if these essential routines are not available
+                    continue
+
+                target_tx_disable_flag = data['host_tx_ready'] == 'false'
```

Comment:
Prince suggested: _subscribes to admin_status as well, to make sure only turn on TX if both admin_status and host_tx_ready are true_

Reply:
added admin_status check